realm/src/main/java/io/realm/internal/CheckedRow.java

```diff
@@ -86,6 +86,29 @@ public boolean isNullLink(long columnIndex) {
         }
     }
 
+    @Override
+    public boolean isNull(long columnIndex) {
+        ColumnType columnType = getColumnType(columnIndex);
+        return super.isNull(columnIndex);
+    }
+
+    @Override
+    public void setNull(long columnIndex) {
+        ColumnType columnType = getColumnType(columnIndex);
+        if (columnType == ColumnType.STRING) {
+            super.setString(columnIndex, null);
```

Comment:
this String column might be required? shouldn't we check using `isColumnNullable` before? same for binary